File: Common Algorithm/Sorting Algorithm/SelectionSort/selection_sort.py
def selection_sort(arr):
    n = len(arr)

    for i in range(n - 1):
        minIndex = i
        for j in range(i + 1, n):
            if arr[j] < arr[minIndex]:
                minIndex = j
        
        arr[i], arr[minIndex] = arr[minIndex], arr[i]

given_arr = [2, 5, 1, 6, 8]
selection_sort(given_arr)
print(given_arr)

File: Common Algorithm/Sorting Algorithm/SelectionSort/selection_sort.js
let givenArr = [3, 2, 5, 8, 1];

function selectionSort(arr) {
  const n = arr.length;

  for (let i = 0; i < n - 1; i++) {
    let minIndex = i;

    for (let j = i + 1; j < n; j++) {
      if (arr[j] < arr[minIndex]) minIndex = j;
    }

    if (i !== minIndex) [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]];
  }
}

selectionSort(givenArr);

console.log(givenArr);
